sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs

```diff
@@ -10,18 +10,36 @@
 
 namespace Azure.Identity
 {
+    /// <summary>
+    /// Enables authentication of a service principal in to Azure Active Directory using a X509 certificate that is assigned to it's App Registration. More information
+    /// on how to configure certificate authentication can be found here:
+    /// https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-certificate-credentials#register-your-certificate-with-azure-ad
+    /// </summary>
     public class ClientCertificateCredential : TokenCredential
     {
         private string _tenantId;
         private string _clientId;
         private X509Certificate2 _clientCertificate;
         private AadIdentityClient _client;
 
+        /// <summary>
+        /// Creates an instance of the ClientCertificateCredential with the details needed to authenticate against Azure Active Directory with the specified certificate.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="clientCertificate">The authentication X509 Certificate of the service principal</param>
         public ClientCertificateCredential(string tenantId, string clientId, X509Certificate2 clientCertificate)
             : this(tenantId, clientId, clientCertificate, null)
         {
         }
 
+        /// <summary>
+        /// Creates an instance of the ClientCertificateCredential with the details needed to authenticate against Azure Active Directory with the specified certificate.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="clientCertificate">The authentication X509 Certificate of the service principal</param>
+        /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
```

Comment:
Consider adding a reference to `IdentityClientOptions` so people know what the options are

Reply:
The type will be linked in the generated docs.